util/platform/netdev/NetPlatform_linux.c

```diff
@@ -15,92 +15,204 @@
 #include "util/platform/netdev/NetPlatform.h"
 #include "util/platform/Sockaddr.h"
 
-#include <string.h>
-#include <errno.h>
 #include <stdio.h>
-#include <sys/ioctl.h>
-#include <fcntl.h>
+#include <errno.h>
 #include <unistd.h>
-#include <sys/socket.h>
-#include <sys/types.h>
 #include <stdlib.h>
 #include <stddef.h>
 #include <net/if.h>
 #include <arpa/inet.h>
 #include <linux/if.h>
+#include <linux/if_link.h>
+#include <linux/rtnetlink.h>
+#include <libmnl/libmnl.h>
```

Comment:
So cjdns will nolonger build unless people install libmnl ?  That's going to break auto-updaters...